stack.py

```diff
@@ -8,10 +8,10 @@ def push(self, value):
         self._storage.push(value)
 
     def pop(self):
-        return self._storage.pop()
-
-    def size(self):
-        return len(self._storage)
+        try:
+            return self._storage.pop()
+        except ValueError as exception:
+            raise ValueError("Cannot pop from an empty stack") from exception
 
     def __len__(self):
-        return self.size()
\ No newline at end of file
+        return self._storage._size
```

Comment:
`_size` is a __private__ property and should not be used.
```suggestion
        return len(self._storage)
```